contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -0,0 +1,61 @@
+# Feature gate progression in Kubernetes
+
```

Comment:
bits from https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation might be helpful to repeat or link to from here.

> Feature gates are intended to cover the development life cycle of a feature - they are not intended to be long-term APIs. As such, they are expected to be deprecated and removed after a feature becomes GA or is dropped.

and 

> Optional features intended to be enabled or disabled by users even once the feature is GA should include a mechanism for enabling or disabling the feature (like a command-line flag or config file option) in addition to the associated feature gate.